### PR TITLE
Feature/extend docker

### PR DESCRIPTION
**Pull request by williamayerst**
Pull Request Template

📲 What

Adds Write-Information lines to Build-DockerImage.

🤔 Why

To assist troubleshooting and increase visibility of build process.

🛠 How

Standard Write-Information PS implementation

👀 Evidence

https://amido-dev.visualstudio.com/Amido-Stacks/_build/results?buildId=20245&view=results

Example of the information being exported `Running: docker push docker.io/williamamido/runner-pwsh-dotnet:latest`

🕵️ How to test

Run `Build-DockerImage` with the relevant environment variables set i.e. DOCKER_USERNAME, DOCKER_PASSWORD, etc.  - no other changes are required.

### CODE REVIEW

File: src/modules/AmidoBuild/exported/Build-DockerImage.ps1 [hunk 1]
```diff
@@ -132,16 +133,24 @@ function Build-DockerImage() {
...
         }
 
-        # Finally push the image
+        # Push the image with the desired tag
         $cmd = "docker push {0}/{1}:{2}" -f $registry, $name, $tag
+        Write-Information -MessageData ("Running: {0}" -f $cmd)
         Invoke-External -Command $cmd
 
+        # Push the image with the latest tag
+        $cmd = "docker push {0}/{1}:latest" -f $registry, $name
```
Comment (ElvenSpellmaker) [inline src/modules/AmidoBuild/exported/Build-DockerImage.ps1:149]: Like previous Stacks IMHO this should be optional, default off.

Comment (williamayerst) [reply src/modules/AmidoBuild/exported/Build-DockerImage.ps1]: Added with `-latest` parameter, tests updated @ElvenSpellmaker

File: src/modules/AmidoBuild/exported/Build-DockerImage.Tests.ps1 [hunk 2]
```diff
@@ -139,4 +139,31 @@ Describe "Build-DockerImage" {
...
+
+            # Ensure that the image is pushed to the registry with specific tag
+            $Session.commands.list[2] | Should -BeLike "*docker* push docker.io/pester-tests:unittests"
+
+            # Ensure that the image is pushed to the registry with latest tag
+            $Session.commands.list[3] | Should -BeLike "*docker* push docker.io/pester-tests:latest"
+        }
+    }
+
 }
```
Comment (ElvenSpellmaker) [inline src/modules/AmidoBuild/exported/Build-DockerImage.Tests.ps1:169]: Newline missing, you might want to configure your IDE to add them

Comment (ElvenSpellmaker) [reply src/modules/AmidoBuild/exported/Build-DockerImage.Tests.ps1:169]: Note: Maybe some form of lining around it, but not sure the best way currently.

File: src/modules/AmidoBuild/exported/Build-DockerImage.ps1 [hunk 3]
```diff
@@ -132,16 +138,27 @@ function Build-DockerImage() {
...
-        # Finally push the image
+        # Push the image with the desired tag
         $cmd = "docker push {0}/{1}:{2}" -f $registry, $name, $tag
+        Write-Information -MessageData ("Running: {0}" -f $cmd)
         Invoke-External -Command $cmd
 
+        # Push the image with the latest tag if latest flag is declared
+        if ($latest.IsPresent) {
+            $cmd = "docker push {0}/{1}:latest" -f $registry, $name
+            Write-Information -MessageData ("Running: {0}" -f $cmd)
```
Comment (ElvenSpellmaker) [inline src/modules/AmidoBuild/exported/Build-DockerImage.ps1:156]: This whole cmd --> Write-Information --> Invoke-External is repeated at least one (and potentially in other files?)
I wonder if we need a method that takes a cmd string and does both lines in one?

File: src/modules/AmidoBuild/exported/Build-DockerImage.Tests.ps1 (hunk 2 above)
Comment (williamayerst) [reply src/modules/AmidoBuild/exported/Build-DockerImage.Tests.ps1]: Done and done, cheers!

File: src/modules/AmidoBuild/exported/Build-DockerImage.ps1 (hunk 3 above)
Comment (williamayerst) [reply src/modules/AmidoBuild/exported/Build-DockerImage.ps1:156]: Good shout - amending Invoke-External to include this block instead?

**Pull request merged**